extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLSubscriptionHandler.java

```diff
@@ -0,0 +1,118 @@
+                public void handle(AsyncResult<ServerWebSocket> event) {
+                    if (event.succeeded()) {
+                        ServerWebSocket socket = event.result();
+
+                        socket.textMessageHandler(new Handler<String>() {
+                            @Override
+                            public void handle(String message) {
+                                final AtomicReference<Subscription> subscriptionRef = new AtomicReference<>();
+                                JsonObject jsonInput = inputToJsonObject(message);
+                                ExecutionResponse executionResponse = getExecutionService()
+                                        .execute(jsonInput);
+
+                                ExecutionResult executionResult = executionResponse.getExecutionResult();
+                                if (executionResult != null) {
+
+                                    // If there is error on the query, we can not start a subscription
+                                    if (executionResult.getErrors() != null && !executionResult.getErrors().isEmpty()) {
+                                        socket.writeTextMessage(executionResponse.getExecutionResultAsString());
+                                        socket.close();
+                                    } else {
+                                        Publisher<ExecutionResult> stream = executionResponse.getExecutionResult().getData();
+
+                                        if (stream != null) {
+                                            stream.subscribe(new Subscriber<ExecutionResult>() {
+
+                                                @Override
+                                                public void onSubscribe(Subscription s) {
+                                                    subscriptionRef.set(s);
+                                                    s.request(1);
+                                                }
+
+                                                @Override
+                                                public void onNext(ExecutionResult er) {
+                                                    Object response = er.getData();
+                                                    if (!socket.isClosed()) {
+                                                        socket.writeTextMessage(JSONB.toJson(response));
+                                                    }
+
+                                                    Subscription s = subscriptionRef.get();
+                                                    s.request(1);
+                                                }
+
+                                                @Override
+                                                public void onError(Throwable t) {
+                                                    if (!socket.isClosed()) {
+                                                        socket.writeTextMessage(t.getMessage());
```

Comment:
This should probably
- include the whole stack trace if the exception is in the show list (?)
- close the websocket connection because there can be no more events after this

Reply:
This is not a normal GraphQL error, those are in the response (now :) ).  So not sure when this will happen. For now let's keep it like this, until I find a way to simulate this. Ok ?

Reply:
This happens when you have a failing Multi:
```
@Subscription
public Multi<Person> getNewPeople() {
   return Multi.createFrom().failure(new RuntimeException("blabla"));
}
```

A failing multi is a relatively common stuff that we should account for. And once a Multi produces a failure, it can't continue producing items.

Reply:
Ok cool. Let me have a look . Thanks :)

Reply:
@jmartisk - a proper fix for this would be in SmallRye, so that all the rules on should we display the message etc applies. For now I added it in the extension, with a TODO to remove once we have a new release of SmallRye. I'll do another release of SmallRye before Quarkus 2. Let me know.